From 738c520300d11d336d0c406561da82c9010e315d Mon Sep 17 00:00:00 2001
From: Richard van der Hoff <git@rvanderhoff.org.uk>
Date: Wed, 26 Jun 2013 22:51:19 +0100
Subject: [PATCH 09/20] remove a couple of bits of debug code

Poll for jobs before making xmpp connection, so that basic functionality still
works, even when xmpp auth fails
---
 cloudprint/cloudprint.py | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index e5ec92b..b8dff08 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -21,8 +21,6 @@ import xmpp
 XMPP_SERVER_HOST = 'talk.google.com'
 XMPP_USE_SSL = True
 XMPP_SERVER_PORT = 5223
-#XMPP_SERVER_HOST = 'localhost'
-#XMPP_USE_SSL = False
 
 SOURCE = 'Armooo-PrintProxy-1'
 PRINT_CLOUD_SERVICE_ID = 'cloudprint'
@@ -189,7 +187,6 @@ class CloudPrintProxy(object):
             LOGGER.info('Added Printer ' + name)
 
     def update_printer(self, printer_id, name, description, ppd):
-        return
         r = self.get_rest()
         r.post(
             PRINT_CLOUD_URL + 'update',
@@ -357,15 +354,15 @@ def process_jobs(cups_connection, cpp, printers):
 
     while True:
         try:
-            if not xmpp_conn.isConnected():
-                xmpp_conn.connect(XMPP_SERVER_HOST,XMPP_SERVER_PORT,
-                                  XMPP_USE_SSL,xmpp_auth)
-
             for printer in printers:
-                 for job in printer.get_jobs():
+                for job in printer.get_jobs():
                     process_job(cups_connection, cpp, printer, job)
             sleeptime = POLL_PERIOD
 
+            if not xmpp_conn.isConnected():
+                xmpp_conn.connect(XMPP_SERVER_HOST,XMPP_SERVER_PORT,
+                                  XMPP_USE_SSL,xmpp_auth)
+
             xmpp_conn.awaitNotification(sleeptime)
 
         except Exception:
-- 
1.8.4.rc3

